pkg/client/core/state/storm/argocd_state.go

```diff
@@ -45,17 +46,37 @@ func (a *ArgoCD) Convert() *client.ArgoCD {
 }
 
 func (a *argoCDState) SaveArgoCD(cd *client.ArgoCD) error {
-	return a.node.Save(NewArgoCD(cd, NewMetadata()))
+	existing, err := a.getArgoCD()
```

Comment:
Would it not be possible to create a single implementation for the repeated logic used in all the x_state.go files? Or will the interface{} start spreading too much perhaps?

Reply:
Yes, I think this should be possible, I am simply erring on the side of caution for now, in general I have found that generalising too early can cause more pain than it is worth